docs/user-guide/spark-user-guide.md

```diff
@@ -0,0 +1,159 @@
+
+An example cql-jobs.json file is show below...
+```json
+{
+	"globalParameters": {
+		"Measurement Period": {
+			"type": "interval",
+			"start": {
+				"type": "date",
+				"value": "2020-01-01"
+			},
+			"end": {
+				"type": "date",
+				"value": "2021-01-01"
+			}
+		}
+	},
+	"evaluations": [
+		{
+			"descriptor": {
+				"libraryId": "SampleLibrary",
+				"version": "1.0.0"
+			},
+			"expressions": [
+				"IsFemale"
+			],
+			"parameters": {
+				"MinimumAge": {
+					"type": "integer",
+					"value": 17
+				}
+			},
+			"contextKey": "Patient",
+			"contextValue": "NA"
+		}
+	]
+}
+```
+
+Parameters can be expressed as any type on the CQL System typesystem. The types available and the encoding format for the more complex datatypes is described in the [Parameter Formats](https://alvearie.io/quality-measure-and-cohort-service/#/user-guide/parameter-formats) page of the user guide. 
+
+If the user wishes to limit which jobs are evaluated during a specific application run, there are program options available that limit which libraries (`-l`) and expressions (`-e`) are evaluated. For the purposes of job filtering, cql library version is ignored. If only the `-e` option is provided, then all expressions with that name will be evaluated across all cql job definitions. 
+
+### Program Output
+
+For each aggregation context, an separate output table is created. The storage path for the output table is configured using `-o` options as described above. The records of the context output table consist of the unique key for the specific context record (sharing the same name as the field in the model definition) and a column for each CQL expression that is evaluated. The default column name for the CQL evaluation columns is "<Library Name>|<Expression Name>". The hyphen delimiter can be changed using the `--default-output-column-delimiter` program option or users can provide their own column name alongside the expression in the `cql-jobs.json` file. If more than one column ends up with the same name, a runtime error will be thrown.
```

Comment:
It wasn't clear to me how you can express a custom output column name alongside an expression in cql-jobs.json. Maybe add that to the example and elaborate a bit more on how that's done?

Reply:
I was intentionally fudging here because that work is still in progress. I debated leaving it out entirely, but I suspect it will be there by the time HI actually starts using our solution. The actual work will probably involve a change to cql-jobs.json that looks like this... `expresssions: [ { "name": 'MyDefine', outputColumn: 'OutputColumnName' } ]` but I wasn't willing to commit to that until Matt finished the work.

Reply:
The only other option I thought of is a separate field with mappings. Something like `outputColumnOverride: [{"expression1": "outputName1"}, {"expression2": "outputName2"}]`. Main advantage would be that existing tests and cql jobs files should be backwards compatible with adding a completely separate field that controls output name overrides. That being said, I think I like the logical grouping that you get from using the example format that Corey shared.

Does anyone have strong preferences one way or another?

Reply:
I was thinking about backwards compatibility myself and I wonder if we could implement a custom deserializer for that field that understands how to deal with both Strings and Maps? Seems plausible.

In the same vein, I don't really like putting outputColumnName in the CqlEvaluationRequest object because outputColumnName is specific to the Spark application and my original design goal was to keep that object generic enough to be used elsewhere, such as if/when we update the CLI to use the cql-evaluator project. If the fields were separate, you could view that as an extension to the base object  (e.g. SparkCqlEvaluationRequest) .

Would we ever put the column name overrides as a single top-level element in CqlEvaluationRequests instead of CqlEvaluationRequest? Or does it belong in a completely separate overrides file? I'm not sure either is a good idea from a usability standpoint, but I like the idea of having all of the overrides for a single context in a single place because it helps make it clear that they are all related and that column names need to be unique.

Reply:
> Would we ever put the column name overrides as a single top-level element in CqlEvaluationRequests instead of CqlEvaluationRequest?

I'd have to think about this a bit. I think  if we separated out the overrides completely, then we'd need a unique name/id per CqlEvaluationRequest so we could tie overrides to a particular request.